### ISSUE TITLE
Docker image may be unavailable when starting a unit, leading to overly slow startup

### ISSUE THREAD

**Issue by itamarst** [ClusterHQ/flocker#350]
**Replaced by** https://clusterhq.atlassian.net/browse/FLOC-350

`systemd` gets annoyed and times out if starting a unit takes too long (see e.g. transcript in #326). Starting a unit requires starting a Docker container which may involve a long download of a Docker image. This is also an availability issue insofar as it means that sometimes when an application is moved it can take quite a while for it to start back up.

To prevent this timeout we should pull the Docker image before starting the container. This is not sufficient to solve the availability issue (that requires coordinating between nodes so the origin node only stops the application once the destination node finishes a pull), so a followup issue should be opened once this one is resolved.

**Comment by adamtheturtle** [comment]
We are moving our development planning to JIRA. This issue is now being tracked at https://clusterhq.atlassian.net/browse/FLOC-350. You are welcome to file additional issues in GitHub if that's easier for you.
